internal/service/iam/role_test.go

```diff
@@ -699,6 +699,24 @@ func TestAccIAMRole_InlinePolicy_empty(t *testing.T) {
 	})
 }
 
+func TestAccIAMRole_InlinePolicy_malformed(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.IAMServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRoleDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccRoleConfig_policyInlineMalformed(rName, rName),
+				ExpectError: regexache.MustCompile(`MalformedPolicyDocument`),
```

Comment:
This is a little tough to test fully because `Check` doesn't run if an error is encountered. I would like to do a `testAccCheckRoleDestroy` to make sure it's gone before the destroy phase, but, I verified by tracing execution that the role is deleted in this case.